### PR TITLE
feat: restore code to generate docs from a package name

### LINKED ISSUE #1872: Bring back `dagger doc`
## Problem

Dagger 0.1 had a command called `dagger doc`, which was similar to `go doc`. Given the names of a CUE package and definition, it would print information about their API.

Unfortunately, this feature was not compatible with Dagger 0.2, because it relied on concepts such as `@dagger(input)` and `@dagger(output)` which no longer exist. This is also why we currently don't have an API reference in our docs: the generator has not been ported to 0.2.

As a result, there is no easy way to look up available packages and their APIs while developing a Dagger 0.2 plan.

## Solution

Bring back `dagger doc` in a simpler form. Instead of scanning the CUE code and generating documentation, just print the CUE code as is. The output can be refined later - but at least developers get the equivalent of looking up the source file on github, which is what they need to do anyway. It's a strictly better developer experience, achievable in the short term.

### PR DESCRIPTION
**Pull request by samalba**
This PR restores the `dagger doc [package name]`

The whole lib generation works partially (disabled on purpose).

Closes #1872 

cc @dolanor

### CODE REVIEW

**Comment by mircubed** [comment]
@TomChv since Andrea and Solomon are on PTO this week, can we have someone else do the second approval?

**Comment by samalba** [comment]
@TomChv (or other maintainers) I'll let you do some tests on it before merging. Feel free to merge directly if it looks good to you after you run your tests.

**Comment by TomChv** [comment]
Doing some test on your PR.

**TL;DR**: It works well but there are minors issues, I think we can fix them later, it's already useful but as experimental command IMO.

#### Simple `dagger doc` ✅ 

`dagger doc`: throw an error to ask for a special package

> That would be useful to have a command `dagger doc list` to see which packages is available or directly display list of packages

#### Universe package ✅ 

`dagger doc universe.dagger.io/docker`: Works well, display required fields correctly.
However, for `docker.#Run`, there is no required field so nothing is displayed. 
It's because everything is conditional in this package. Honestly I'm not sure we can fix it right now, we should track this one in an issue.

```
docker.#Run

    Run a command in a container
```

That make me think that we should certainly enable `dagger doc` only as `experimental` one since it's not production ready?
WDYT?

#### Nested package ✅ 

`dagger doc universe.dagger.io/aws/cli`: that's good but `_I_` is displayed, I'm not sure it's relevant.

```
cli.#Command

    Command provides a declarative interface to the AWS CLI
        result       _|_     -

```

#### Core 🟧 

`dagger doc dagger.io/dagger/core`: Great display but we have a warning for `#Mount` type

```shell
11:01PM WARN  system | cannot get field values, ignoring field: #Mount: non-concrete value struct    fieldName=#Mount
```

#### Alpha ✅ 

`dagger doc universe.dagger.io/alpha/rust`: No problem

**Comment by samalba** [comment]
Thanks @TomChv for the tests. I think some of those issues can be addressed in a follow-up PR.

**Review by TomChv — APPROVED** [review]

**Comment by TomChv** [comment]
It's approved on my side, may you can wait for another one to be sure that I did not fort get anything

**Comment by jpadams** [comment]
Could be very useful (esp markdown) for us in onboarding/universe marketplace/catalog.

The fact that nothing I've tried has an `Input` or `Output` makes me thing it's the `0.1` notion of inputs/outputs.

image

I'd like to see the required/optional (non-hidden/private) fields listed.

I'd also like an option to expose fields of embedded definitions (like the `docker.#Run` fields inside something else) so you can see what you could override.

Seems like some authors have embedded markdown in their packages (`yarn`) for instance. But it seems to work ok in `-f md` mode.
image

Seems to be some logic built into markdown render for our docs site since their are special `sidebar_label` sections:

image

**Comment by samalba** [comment]
> The fact that nothing I've tried has an `Input` or `Output` makes me think it's the `0.1` notion of inputs/outputs.

Indeed, the notion of inputs/outputs was removed with this changeset. And I have a very different output with yarn. Are you sure you rebuilt the binary when you tried?
